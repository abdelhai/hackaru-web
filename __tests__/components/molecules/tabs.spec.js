import Factory from '@/__tests__/__setups__/factory';
import Tabs from '@/components/molecules/tabs';

describe('Tabs', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(Tabs, {
      propsData: {
        items: ['Home', 'Reports', 'Calendar']
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when click item', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper
        .findAll('li')
        .at(1)
        .trigger('click');
    });

    it('emit change', () => {
      expect(wrapper.emitted('change')[0]).toEqual([1]);
    });
  });
});
