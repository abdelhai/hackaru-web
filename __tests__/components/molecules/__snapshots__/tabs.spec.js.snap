// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs render correctly 1`] = `
<ul>
  <li
    class="selected"
  >
    
    Home
  
  </li>
  <li
    class=""
  >
    
    Reports
  
  </li>
  <li
    class=""
  >
    
    Calendar
  
  </li>
</ul>
`;
